pkg/entrypoint/run_test.go

```diff
@@ -0,0 +1,131 @@
+// 			args:           []string{"sleep", "10"},
+// 			timeout:        1 * time.Second,
+// 			gracePeriod:    1 * time.Second,
+// 			expectedLog:    "level=error msg=\"Process did not finish before 1s timeout\" \nlevel=error msg=\"Process gracefully exited before 1s grace period\" \n",
+// 			expectedMarker: strconv.Itoa(InternalErrorCode),
+// 		},
+// 		{
+// 			name:           "command times out and ignores interrupt",
+// 			args:           []string{"bash", "-c", "trap 'sleep 10' EXIT; sleep 10"},
+// 			timeout:        1 * time.Second,
+// 			gracePeriod:    1 * time.Second,
+// 			expectedLog:    "level=error msg=\"Process did not finish before 1s timeout\" \nlevel=error msg=\"Process did not exit before 1s grace period\" \n",
+// 			expectedMarker: strconv.Itoa(InternalErrorCode),
+// 		},
+// 	}
+
+// 	// we write logs to the process log if wrapping fails
+// 	// and cannot write timestamps or we can't match text
+// 	logrus.SetFormatter(&logrus.TextFormatter{DisableTimestamp: true})
+
+// 	for _, testCase := range testCases {
+// 		t.Run(testCase.name, func(t *testing.T) {
+// 			tmpDir, err := ioutil.TempDir("", testCase.name)
+// 			if err != nil {
+// 				t.Errorf("%s: error creating temp dir: %v", testCase.name, err)
+// 			}
+// 			defer func() {
+// 				if err := os.RemoveAll(tmpDir); err != nil {
+// 					t.Errorf("%s: error cleaning up temp dir: %v", testCase.name, err)
+// 				}
+// 			}()
+
+// 			options := Options{
+// 				Args:        testCase.args,
+// 				Timeout:     testCase.timeout,
+// 				GracePeriod: testCase.gracePeriod,
+// 				Options: &wrapper.Options{
+// 					ProcessLog:            path.Join(tmpDir, "process-log.txt"),
+// 					MarkerFile:            path.Join(tmpDir, "marker-file.txt"),
+// 					ShouldWaitForPrevStep: false,
+// 					PreRunFile:            path.Join(tmpDir, "0"),
+// 					PostRunFile:           path.Join(tmpDir, "0"),
+// 				},
+// 			}
+
+// 			if code := strconv.Itoa(options.Run()); code != testCase.expectedMarker {
+// 				t.Errorf("%s: exit code %q does not match expected marker file contents %q", testCase.name, code, testCase.expectedMarker)
+// 			}
+
+// 			compareFileContents(testCase.name, options.ProcessLog, testCase.expectedLog, t)
+// 			compareFileContents(testCase.name, options.MarkerFile, testCase.expectedMarker, t)
+// 			if options.ShouldWaitForPrevStep {
+// 				compareFileContents(testCase.name, options.PreRunFile,
+// 					testCase.expectedPreRunFile, t)
+// 			}
+// 			if options.ShouldRunPostRun {
+// 				compareFileContents(testCase.name, options.PostRunFile,
+// 					testCase.expectedPostRunFile, t)
+// 			}
+// 		})
+// 	}
+// }
+
+// func compareFileContents(name, file, expected string, t *testing.T) {
+// 	data, err := ioutil.ReadFile(file)
+// 	if err != nil {
+// 		t.Fatalf("%s: could not read file: %v", name, err)
+// 	}
+// 	if string(data) != expected {
+// 		t.Errorf("%s: expected contents: %q, got %q", name, expected, data)
+// 	}
```

Comment:
was it intentional to comment this out? if so, let's delete it

Reply:
the tests are added back now

Reply:
I don't see this in the latest commit.

Reply:
it should be there now